rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c

```diff
@@ -69,13 +70,25 @@ static void detect_driver_type(void)
  * in enclave range.
  */
 static int create_enclave_range(int dev_fd, uint64_t mmap_size,
-				struct enclave_info *encl_info)
+				struct enclave_info *encl_info,
+				struct metadata *meta_data)
 {
 	uint64_t mmap_min_addr;
 
 	if (get_mmap_min_addr(&mmap_min_addr) < 0)
 		return -1;
 
+	/* mmap_min_addr only affect the value of encl_offset when
+	 * in tree driver with null_dereference_protection.
+	 */
+	if ((mmap_min_addr != meta_data->mmap_min_addr) && !is_oot_driver &&
+	    meta_data->null_dereference_protection) {
```

Comment:
why to judge mmap_min_addr rather than meta_data->mmap_min_addr?

Reply:
mmap_min_addr is cac

Reply:
I print the right `mmap_min_addr` in fprintf